internal/ingress/annotations/wallarm/main.go

```diff
@@ -18,25 +18,160 @@ limitations under the License.
+func validateParserDisable(s string) error {
+	allowedParsers := map[string]bool{
+		"cookie":    true,
+		"zlib":      true,
+		"htmljs":    true,
+		"json":      true,
+		"multipart": true,
+		"base64":    true,
+		"percent":   true,
+		"urlenc":    true,
+		"xml":       true,
+		"jwt":       true,
+	}
+	for _, value := range strings.Split(s, ",") {
+		parser := strings.TrimSpace(value)
+		if _, ok := allowedParsers[value]; !ok {
+			return fmt.Errorf("unknown parser \"%s\"", parser)
+		}
+	}
+	return nil
+}
+
+var wallarmAnnotations = parser.Annotation{
+	Annotations: parser.AnnotationFields{
+		wallarmModeAnnotation: {
+			Validator: parser.ValidateOptions(
+				[]string{"off", "monitoring", "safe_blocking", "block"}, true, true,
+			),
+			Scope:         parser.AnnotationScopeLocation,
+			Risk:          parser.AnnotationRiskLow,
+			Documentation: `Traffic processing mode`,
+		},
+		wallarmModeAllowOverrideAnnotation: {
+			Validator: parser.ValidateOptions(
+				[]string{"off", "strict", "on"}, true, true,
+			),
+			Scope:         parser.AnnotationScopeLocation,
+			Risk:          parser.AnnotationRiskLow,
+			Documentation: `Manages the ability to override the wallarm_mode values via filtering rules downloaded from the Wallarm Cloud (custom ruleset)`,
+		},
+		wallarmFallbackAnnotation: {
+			Validator: parser.ValidateOptions(
+				[]string{"off", "on"}, true, true,
+			),
+			Scope:         parser.AnnotationScopeLocation,
+			Risk:          parser.AnnotationRiskLow,
+			Documentation: `With the value set to on, NGINX has the ability to enter an emergency mode; if proton.db or custom ruleset cannot be downloaded, this setting disables the Wallarm module for the http, server, and location blocks, for which the data fails to download. NGINX keeps functioning`,
+		},
+		wallarmApplicationAnnotation: {
+			AnnotationAliases: []string{wallarmInstanceAnnotation},
+			Validator:         validateApplicationId,
+			Scope:             parser.AnnotationScopeLocation,
+			Risk:              parser.AnnotationRiskLow,
+			Documentation:     `Unique identifier of the protected application to be used in the Wallarm Cloud. The value can be a positive integer except for 0`,
+		},
+		wallarmPartnerClientUuidAnnotation: {
+			Validator:     parser.ValidateRegex(regexp.MustCompile(`[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}`), true),
+			Scope:         parser.AnnotationScopeLocation,
+			Risk:          parser.AnnotationRiskLow,
+			Documentation: `Unique identifier of the tenant for the multi-tenant Wallarm node. The value should be a string in the UUID format`,
+		},
+		// FIXME: ?
+		wallarmBlockPageAnnotation: {
+			Validator:     func(string) error { return nil },
+			Scope:         parser.AnnotationScopeLocation,
+			Risk:          parser.AnnotationRiskLow,
+			Documentation: `https://docs.wallarm.com/admin-en/configure-parameters-en/#wallarm_block_page`,
+		},
+		// FIXME: ?
+		wallarmAclBlockPageAnnotation: {
```

Comment:
This is not even mentioned in https://docs.wallarm.com/admin-en/configure-kubernetes-en/#ingress-annotations